### PR TITLE
MCR-2333-commands_to_process_commands_file_from_Classpath

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2333).

### CODE REVIEW

**Review by rsteph-de — APPROVED** [review]
looks good for me

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on be03ed39198493cfe7508d56f806edb851b3ccc2! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6357-pull_request-1137

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on cbcd5dda0542614d73defc286462c199cc4079c8! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6359-pull_request-1137

File: mycore-webcli/src/main/java/org/mycore/webcli/cli/MCRBasicWebCLICommands.java [hunk 1]
```diff
@@ -30,24 +31,31 @@
 
 @MCRCommandGroup(name = "Basic commands")
 public class MCRBasicWebCLICommands {
-    @MCRCommand(syntax = "process {0}", help = "Execute the commands listed in the text file {0}.")
+
+    @MCRCommand(syntax = "process from resources {0}", help = "Execute the commands listed in the resource file {0}.",
```
Comment (yagee-de) [inline mycore-webcli/src/main/java/org/mycore/webcli/cli/MCRBasicWebCLICommands.java]: "process from resources foo/bar/setup.txt" --> "process resource foo/bar/setup.txt"

File: mycore-webcli/src/main/java/org/mycore/webcli/cli/MCRBasicWebCLICommands.java [hunk 2]
```diff
@@ -30,24 +31,31 @@
...
+            order = 10)
+    public static List<String> readCommandsResource(String resource) throws IOException, URISyntaxException {
```
Comment (yagee-de) [inline mycore-webcli/src/main/java/org/mycore/webcli/cli/MCRBasicWebCLICommands.java]: URISysntaxException is never thrown

File: mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java [hunk 3]
```diff
@@ -321,19 +325,35 @@ public static void getURI(String uri, String file) throws Exception {
...
         try (BufferedReader reader = Files.newBufferedReader(new File(file).toPath(), Charset.defaultCharset())) {
             output("Reading commands from file " + file);
+            return readCommandsFromBufferedReader(reader);
+        }
+    }
+
+    public static List<String> readCommandsRessource(String resource) throws IOException {
+        final URL resourceURL = MCRClassTools.getClassLoader().getResource(resource);
+        if (resourceURL == null) {
+            return Collections.emptyList();
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java]: Should throw an IOException here if the resource is not found.

File: mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java [hunk 4]
```diff
@@ -321,19 +325,35 @@ public static void getURI(String uri, String file) throws Exception {
...
+        }
+        try (InputStream is = resourceURL.openStream()) {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, Charset.defaultCharset()))) {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java]: Better link Autoclosables in one line
`try (InputStream is=..; InputStreamReader ir=...; BufferedReader reader=...)`
Consider NOT using "`defaultCharset()`" for resources but use a defined one like `StandardCharsets.UTF_8`

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java [hunk 5]
```diff
@@ -18,11 +18,23 @@
...
+import org.jdom2.Element;
+import org.mycore.common.MCRClassTools;
+import org.mycore.common.MCRSession;
+import org.mycore.common.MCRSessionMgr;
+import org.mycore.common.MCRSystemUserInformation;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.common.content.MCRJDOMContent;
+import org.mycore.common.events.MCRStartupHandler;
+import org.mycore.common.xml.MCRURIResolver;
+
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java]: please fix import order

File: mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java [hunk 6]
```diff
@@ -332,13 +330,12 @@ public static void getURI(String uri, String file) throws Exception {
     public static List<String> readCommandsRessource(String resource) throws IOException {
         final URL resourceURL = MCRClassTools.getClassLoader().getResource(resource);
         if (resourceURL == null) {
-            return Collections.emptyList();
+            throw new IOException("Resource URL is null!");
         }
-        try (InputStream is = resourceURL.openStream()) {
-            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, Charset.defaultCharset()))) {
+        try (BufferedReader reader = new BufferedReader(
+            new InputStreamReader(resourceURL.openStream(), Charset.defaultCharset()))) {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandLineInterface.java]: may leak resourceURL.openStream() and do not use defaultCharset() for classpath resources but stick it to UTF8 from StandardCharsets class

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Review by yagee-de — APPROVED** [review]

**Pull request merged**